### PR TITLE
[MetaSchedule] Enable Task Filtering

### PR DESCRIPTION
**Pull request by junrushao1994**
This PR allows `relay.backend.MetaScheduleExtractTask` to take an extra argument `filter_func` which filters out unintentional tasks that don't need tuning. The counterpart of AutoScheduler is within `traverse_to_get_io_tensors`.

### CODE REVIEW

**Comment by junrushao1994** [comment]
CC: @comaniac @Hzfengsy @zxybazh

**Comment by Hzfengsy** [comment]
Can we add testcases?

File: src/relay/backend/task_extraction.cc [hunk 1]
```diff
@@ -31,25 +31,57 @@ namespace tvm {
...
-
-using meta_schedule::ExtractedTask;
+bool DefaultTaskFilter(const Array<te::Tensor>& args) {
+  using namespace ::tvm::te;
+  std::vector<Tensor> stack;
+  std::unordered_set<const TensorNode*> visited;
+  for (const Tensor& v : args) {
+    for (const PrimExpr& e : v->shape) {
+      if (!e->IsInstance<IntImmNode>()) {
+        return false;
```
Comment (comaniac) [inline src/relay/backend/task_extraction.cc:41]: Better to have some comments in this function, such as "doesn't support dynamic shape", etc.

**Review by comaniac — COMMENTED** [review]
Overall LGTM. Echo @Hzfengsy's comment that we may need a test case.

File: src/relay/backend/task_extraction.cc (hunk 1 above)
Comment (junrushao1994) [reply src/relay/backend/task_extraction.cc:42]: done!

**Review by comaniac — APPROVED** [review]
IICU, unit tests for task extraction don't need to run tune. Instead, it only needs to extract tasks and verify their number and workloads. However, if MetaSchedule doesn't have unit tests for task extractions yet, we could of course add it later.

**Comment by junrushao1994** [comment]
I'm happy to add a simple unittest without tuning :-)

**Pull request merged**